### PR TITLE
[1LP][RFR] Added test test_widgets_reorder_in_reports

### PR DESCRIPTION
**Pull request by quarckster**
{{pytest: -v -k "test_menuwidget_crud or test_widgets_reorder_in_reports"}}

### CODE REVIEW

**Comment by rlbabyuk** [comment]
@quarckster why we need to move those widgets outside widgetastic_manageiq? I'm just not sure fully understand the case.

**Comment by quarckster** [comment]
@rlbabyuk because they are specific only for that part of the UI

**Comment by rlbabyuk** [comment]
looks good, however, could you please add docstrings for items which were updated/added

**Comment by dajoRH** [comment]
Lint report for commit 3c44a74e05c14d0d82f9b24cfbbcc774c91c0354:

`cfme/intelligence/reports/dashboards.py`:
- :heavy_exclamation_mark: Line 38:1: **W293** *blank line contains whitespace*

`cfme/intelligence/reports/widgets/menu_widgets.py`:
- :heavy_exclamation_mark: Line 23:1: **W293** *blank line contains whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by rlbabyuk** [comment]
but still, wait for @lkhomenk review, as he is also interested in it

**Comment by lkhomenk** [comment]
@quarckster  same question as @rlbabyuk already asked. your widget doesn't look like some specific widget, you do not override anything . so it could be a problem for anyone to find it in new location. is that really needed to move it out?

File: widgetastic_manageiq/__init__.py [hunk 1]
```diff
@@ -2713,65 +2714,110 @@ def ProviderEntity():  # noqa
 
 
 class DashboardWidgetsPicker(View):
-    """ Represents widgets picker in Dashboard editing screen (Cloud Intel/Reports/Dashobards).
+    """Represents widgets picker in Dashboard editing screen (Cloud Intel/Reports/Dashobards).
```
Comment (mfalesni) [inline widgetastic_manageiq/__init__.py]: Can you fix the `Dashobards` typo? Unless there is a bard singing a song for Dasha in real :)

File: widgetastic_manageiq/__init__.py [hunk 2]
```diff
@@ -2713,65 +2714,110 @@ def ProviderEntity():  # noqa
...
+    def drag_and_drop(self, dragged_widget, dropped_widget):
+        dragged_widget_el = self.dashboard_widgets(dragged_widget).div
+        dragged_middle = self.browser.middle_of(dragged_widget_el)
+        dropped_middle = self.browser.middle_of(self.dashboard_widgets(dropped_widget).div)
+        drop_x = dropped_middle.x
+        # In order to perform a successful drag and drop we should drop a dragged widget a bit above
+        # or below of the dropped widget center. If the dragged widget is above of the dropped one
+        # we should add a small delta to "y" coordinate of the dropped widget middle, otherwise that
+        # delta should be deducted.
+        drop_y = dropped_middle.y + math.copysign(5, dropped_middle.y - dragged_middle.y)
```
Comment (mfalesni) [inline widgetastic_manageiq/__init__.py]: :+1:

File: widgetastic_manageiq/__init__.py [hunk 3]
```diff
@@ -2713,65 +2714,110 @@ def ProviderEntity():  # noqa
...
+    """Represents shortcut picker in Menu Widget editing screen
     (Cloud Intel/Reports/Dashboard Widgets/Menus).
 
+    Args:
+        id (str): id attribute of the root div
+        select_id (str): id attribute for BootstrapSelect
+        names_locator (str): xpath for all elements with shortcuts names
     """
+
+    ROOT = ParametrizedLocator(".//div[@id='{@id}']")
```
Comment (mfalesni) [inline widgetastic_manageiq/__init__.py]: Use `quote` and get rid of the quotes :)

**Pull request merged**